# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/webrtc/signaling/gtest/mediapipeline_unittest.cpp
# Commit: 09db50c8f370
# Full Hash: 09db50c8f370b69f186f3e6b7ce6154a984b40b4
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 9
# Description:
#   Bug 1654112 - Pass codec configs by value/reference instead of pointers. r=ng
#   
#   This patch mainly sets the stage for using codec configs with StateMirroring.
#   It also makes the code a bit simpler, and removes the need to handle the nullptr
#   case.
# ==============================================================================

diff -r d1552c620c11 -r 09db50c8f370 media/webrtc/signaling/gtest/mediapipeline_unittest.cpp
--- a/media/webrtc/signaling/gtest/mediapipeline_unittest.cpp	Wed Aug 11 11:33:21 2021 +0200
+++ b/media/webrtc/signaling/gtest/mediapipeline_unittest.cpp	Wed Apr 14 17:36:40 2021 +0200
@@ -369,7 +369,7 @@
     auto rv = WaitFor(InvokeAsync(call_->mCallThread, __func__, [&] {
       return Promise::CreateAndResolve(
           static_cast<AudioSessionConduit*>(audio_conduit_.get())
-              ->ConfigureSendMediaCodec(&audio_config_),
+              ->ConfigureSendMediaCodec(audio_config_),
           "TestAgentSend::ConfigureSendMediaCodec");
     }));
     EXPECT_EQ(kMediaConduitNoError, rv.unwrap());
@@ -399,8 +399,8 @@
   TestAgentReceive(const RefPtr<SharedWebrtcState>& aSharedState,
                    webrtc::WebRtcKeyValueConfig* aTrials)
       : TestAgent(aSharedState, aTrials) {
-    std::vector<UniquePtr<AudioCodecConfig>> codecs;
-    codecs.emplace_back(new AudioCodecConfig(audio_config_));
+    std::vector<AudioCodecConfig> codecs;
+    codecs.push_back(audio_config_);
 
     using Promise = MozPromise<MediaConduitErrorCode, bool, true>;
     auto rv = WaitFor(InvokeAsync(call_->mCallThread, __func__, [&] {